src/Features/postsSlice.js

```diff
@@ -0,0 +1,318 @@
+        return data
+    } catch (err) {
+        return rejectWithValue(err.response.data);
+    }
+})
+export const getAllBookmarks = createAsyncThunk(`/${namespace}/getBookmarks`, async (_, { rejectWithValue }) => {
+    try {
+        const { data } = await axios.get(`/api/users/bookmark`)
+        return data.bookmarks
+    } catch (err) {
+        return rejectWithValue(err.response.data);
+    }
+})
+
+
+//comments
+
+export const addComment = createAsyncThunk(`/${namespace}/addComment`, async ({ postId, commentData }, { rejectWithValue }) => {
+    console.log("in add comm", { commentData })
+    try {
+        const { data } = await axios.post(`/api/comments/add/${postId}`, { commentData })
+        console.log("COMMENTS DATA", data)
+        return data.posts
+    } catch (err) {
+        return rejectWithValue(err.response.data);
+    }
+
+})
+export const editComment = createAsyncThunk(`/${namespace}/editComment`, async ({ postId, commentId, commentData }, { rejectWithValue }) => {
+    console.log("Comment data", { commentData, postId, commentId })
+    try {
+        const { data } = await axios.post(`/api/comments/edit/${postId}/${commentId}`, { commentData })
+        return data.posts
+    } catch (err) {
+        return rejectWithValue(err.response.data);
+    }
+})
+export const deleteComment = createAsyncThunk(`/${namespace}/deleteComment`, async ({ postId, commentId }, { rejectWithValue }) => {
+    console.log("daacomms", { postId, commentId })
+    try {
+        const { data } = await axios.delete(`/api/comments/delete/${postId}/${commentId}`);
+        console.log(data)
+        return data.posts
+    } catch (err) {
+        return rejectWithValue(err.response.data);
+    }
+})
+
+
+const initialStateValue = {
+    posts: [],
+    bookmarks: [],
+    gifUrlSelected: '',
+    sortedBy: 'newest',
+    postStatus: null,
+    loadCommentStatus: null,
+    likeStatus: { status: '', loadId: null },
+    bookmarkStatus: { status: '', loadId: null },
+}
+
+export const postsSlice = createSlice({
+    name: namespace,
+    initialState: initialStateValue,
+    reducers: {
+        sortPosts: (state, action) => {
+            switch (action.payload) {
+                case "trending":
+                    state.posts = getTrendingPosts(current(state.posts))
+                    state.sortedBy = "trending"
+                    break;
+                case 'newest':
+                    state.posts.sort((a, b) => new Date(b.createdAt) - new Date(a.createdAt))
+                    state.sortedBy = "newest"
+                    break;
+                case 'oldest':
+                    state.posts.sort((a, b) => new Date(a.createdAt) - new Date(b.createdAt))
+                    state.sortedBy = "oldest"
+                    break;
+                default:
+                    return state.posts.sort((a, b) => new Date(b.createdAt) - new Date(a.createdAt))
+            }
```

Comment:
Appreciation: Nice use of switch statement

Reply:
thenks